Bottom: 83ee1e3c2805837772fde845d713ed9d08901dec
Top:    2af11d8f9a2a6de4a1fb337a6f2a53655847e8bd
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2009-11-06 16:41:41 +0000

Refresh of benchmark-no-holes-dicer-perfo

---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index cc5c259..3713904 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -1507,10 +1507,6 @@ ghid_draw_everything (BoxTypePtr drawn_area)
   struct cyl_info cyl_info;
   int reverse_layers;
   int save_show_solder;
-  int solder_group;
-  int component_group;
-  int min_phys_group;
-  int max_phys_group;
 
   extern char *current_color;
   extern Boolean Gathering;
@@ -1534,23 +1530,19 @@ ghid_draw_everything (BoxTypePtr drawn_area)
   if (!global_view_2d && save_show_solder)
     reverse_layers = !reverse_layers;
 
-  solder_group = GetLayerGroupNumberByNumber (max_layer + SOLDER_LAYER);
-  component_group = GetLayerGroupNumberByNumber (max_layer + COMPONENT_LAYER);
-
-  min_phys_group = MIN (solder_group, component_group);
-  max_phys_group = MAX (solder_group, component_group);
-
   memset (do_group, 0, sizeof (do_group));
   for (ngroups = 0, i = 0; i < max_layer; i++) {
+    LayerType *l;
     int group;
     int orderi;
 
     orderi = reverse_layers ? max_layer - i - 1 : i;
 
     // Draw in numerical order when in 3D view
-    group = global_view_2d ? GetLayerGroupNumberByNumber (LayerStack[i]) : orderi;
+    l = global_view_2d ? LAYER_ON_STACK (i) : LAYER_PTR (orderi);
+    group = GetLayerGroupNumberByNumber (global_view_2d ? LayerStack[i] : orderi);
 
-    if (!do_group[group]) {
+    if (/*l->On && */!do_group[group]) {
       do_group[group] = 1;
       drawn_groups[ngroups++] = group;
     }
@@ -1576,9 +1568,7 @@ ghid_draw_everything (BoxTypePtr drawn_area)
     DrawLayerGroup (drawn_groups [i], drawn_area);
 
 #if 1
-    if (!global_view_2d &&
-        drawn_groups[i] > min_phys_group &&
-        drawn_groups[i] <= max_phys_group) {
+    if (!global_view_2d && i > 0) {
       cyl_info.from_layer = drawn_groups[i];
       cyl_info.to_layer = drawn_groups[i - 1];
       cyl_info.scale = gport->zoom;
